src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2272,9 +2272,105 @@ def reindex(
         else:
             return self._reindex_axis_1(labels=labels, **kwargs)
 
+    def is_monotonic_decreasing(self, index: bool) -> "SnowflakeQueryCompiler":
+        """
+        Returns a QueryCompiler containing only a column that checks for monotonically
+        decreasing values.
+
+        Parameters
+        ----------
+        index: bool
+            Whether or not this is called by an `Index` object. If False, it is assumed
+            that this method is called by a `Series` object, and monotonicity is checked
+            for the first data column. If True, monotonicity is checked for the first index
+            column.
+
+        Returns
+        -------
+        SnowflakeQueryCompiler
+            QueryCompiler with column to ascertain whether data is monotonically decreasing.
+        """
+        col_to_check = (
+            self._modin_frame.index_column_snowflake_quoted_identifiers[0]
+            if index
+            else self._modin_frame.data_column_snowflake_quoted_identifiers[0]
+        )
+        (
+            new_qc,
+            _,
+            monotonic_decreasing_snowflake_quoted_identifier,
+        ) = self._add_columns_for_monotonicity_checks(
+            col_to_check=col_to_check, columns_to_add="decreasing"
+        )
+        new_modin_frame = new_qc._modin_frame
+        return SnowflakeQueryCompiler(
+            InternalFrame.create(
+                ordered_dataframe=new_modin_frame.ordered_dataframe.limit(
+                    n=1, sort=False
+                ),
+                data_column_pandas_index_names=new_modin_frame.data_column_pandas_index_names,
+                data_column_pandas_labels=["monotonic_column"],
+                data_column_snowflake_quoted_identifiers=[
+                    monotonic_decreasing_snowflake_quoted_identifier
+                ],
+                index_column_pandas_labels=new_modin_frame.index_column_pandas_labels,
+                index_column_snowflake_quoted_identifiers=new_modin_frame.index_column_snowflake_quoted_identifiers,
+                data_column_types=None,
+                index_column_types=None,
+            )
+        )
+
+    def is_monotonic_increasing(self, index: bool) -> "SnowflakeQueryCompiler":
```

Comment:
A lot of the code is shared between `is_monotonic_increasing` and `is_monotonic_decreasing` here - you combine it into one method and decide whether it's increasing or decreasing through a param passed to it